local/local.go

```diff
@@ -87,13 +102,98 @@ func NewImage(repoName string, dockerClient client.CommonAPIClient, ops ...Image
 		downloadBaseOnce: &sync.Once{},
 	}
 
-	for _, v := range ops {
-		image, err = v(image)
-		if err != nil {
-			return nil, err
-		}
+	image, err = processPreviousImageOption(image, imageOpts.prevImageRepoName, platform, dockerClient)
+	if err != nil {
+		return image, err
+	}
+
+	image, err = processBaseImageOption(image, imageOpts.baseImageRepoName, platform, dockerClient)
+	if err != nil {
+		return image, err
+	}
+
+	image, err = prepareNewWindowsImage(image)
+	if err != nil {
+		return image, err
+	}
+
+	return image, nil
+}
+
+func processPlatformOption(defaultPlatform imgutil.Platform, optionPlatform imgutil.Platform) (imgutil.Platform, error) {
+	if optionPlatform.OS != "" && optionPlatform.OS != defaultPlatform.OS {
+		return imgutil.Platform{}, fmt.Errorf(`invalid os: platform os "%s" must match the daemon os "%s"`, optionPlatform.OS, defaultPlatform.OS)
+	}
+
+	if (optionPlatform != imgutil.Platform{}) {
+		return optionPlatform, nil
 	}
 
+	return defaultPlatform, nil
+}
+
+func processPreviousImageOption(image *Image, prevImageRepoName string, platform imgutil.Platform, dockerClient client.CommonAPIClient) (*Image, error) {
+	if prevImageRepoName == "" {
+		return image, nil
+	}
+
+	if _, err := inspectOptionalImage(dockerClient, prevImageRepoName, platform); err != nil {
+		return image, err
+	}
+
+	prevImage, err := NewImage(prevImageRepoName, dockerClient, FromBaseImage(prevImageRepoName))
+	if err != nil {
+		return image, errors.Wrapf(err, "failed to get previous image '%s'", prevImageRepoName)
+	}
+
+	image.prevImage = prevImage
+
+	return image, nil
+}
+
+func processBaseImageOption(image *Image, baseImageRepoName string, platform imgutil.Platform, dockerClient client.CommonAPIClient) (*Image, error) {
+	if baseImageRepoName == "" {
+		return image, nil
+	}
+	inspect, err := inspectOptionalImage(dockerClient, baseImageRepoName, platform)
+	if err != nil {
+		return image, err
+	}
+
+	image.inspect = inspect
+	image.layerPaths = make([]string, len(image.inspect.RootFS.Layers))
+
+	return image, nil
+}
+
+func prepareNewWindowsImage(image *Image) (*Image, error) {
+	if image.inspect.Os != "windows" || len(image.inspect.RootFS.Layers) > 0 {
+		return image, nil
+	}
+
+	layerReader, err := layer.WindowsBaseLayer()
+	if err != nil {
+		return image, err
+	}
+
+	layerFile, err := ioutil.TempFile("", "imgutil.local.image.windowsbaselayer")
+	if err != nil {
+		return image, errors.Wrap(err, "failed to create temp file")
+	}
+	defer layerFile.Close()
+
+	hasher := sha256.New()
+
+	multiWriter := io.MultiWriter(layerFile, hasher)
+
+	if _, err := io.Copy(multiWriter, layerReader); err != nil {
+		return image, errors.Wrap(err, "failed to copy base layer")
+	}
+
+	diffID := "sha256:" + hex.EncodeToString(hasher.Sum(nil))
+
+	image.inspect.RootFS.Layers = append(image.inspect.RootFS.Layers, diffID)
+	image.layerPaths = append(image.layerPaths, layerFile.Name())
```

Comment:
```suggestion
	image.AddLayerWithDiffID(layerFile.Name(), diffID)
```
I think this is slightly more robust to implementation changes.

Reply:
I only ended up doing this in local, as remote was able to skip the temp file and write directly from WindowsBaseLayer to the registry.